share/readonlyadmin.py

```diff
@@ -0,0 +1,56 @@
+## adopted from http://mike.hostetlerhome.com/blog/2012/11/13/add-a-read-only-role-to-django-admin/
+from django.contrib import admin
+from django.core.exceptions import PermissionDenied
+
+
+class ReadOnlyAdmin(admin.ModelAdmin):
+    """
+    """
+
+    def has_add_permission(self, request, obj=None):
+        """
+        Arguments:
+        - `request`:
+        - `obj`:
+        """
+        return self.__user_is_readonly(request)
+
+    def has_delete_permission(self, request, obj=None):
+        """
+        Arguments:
+        - `request`:
+        - `obj`:
+        """
+        return self.__user_is_readonly(request)
+
+    def get_actions(self, request):
+
+        actions = super(ReadOnlyAdmin, self).get_actions(request)
+
+        if self.__user_is_readonly(request):
+            if 'delete_selected' in actions:
+                del actions['delete_selected']
+
+        return actions
+
+    def change_view(self, request, object_id, extra_context=None):
+
+        if self.__user_is_readonly(request):
```

Comment:
Ah, whoops. I think misunderstood something.
For the moment let's assume that there will never be a point in time that we want to give readonly users access to some fields.
You can save a lot of typing and grief if we every update fields by just getting a list of all fields dynamically, your latter suggestion/example.
Take a look at [this](http://stackoverflow.com/questions/13817525/django-admin-make-all-fields-readonly), `self.model._meta.get_all_field_names()` may not be enough.

Alternatively you could just make `readonly_fields` a "yes-man" list. From what I saw in the django code its only used for `x in self.readonly_fields`.

```python
class SetOfEverything(list):
    def __contains__(self, other):
        return True

'foo' in SetOfEverything()  # True
{} in SetOfEverything()  # True
SetOfEverything()  in SetOfEverything()  # Guess that answers that question
```